Libplanet.Net.Tests/Consensus/ConsensusContext/ConsensusContextNonProposerTest.cs

```diff
@@ -233,18 +230,18 @@ public async void HandleMessageFromHigherHeight()
                     codec.Encode(blockHeightThree.MarshalBlock()),
                     -1));
 
-            ConsensusContext.Contexts[3].StepChanged += (sender, tuple) =>
+            ConsensusContext.Contexts[3].StateChanged += (sender, state) =>
             {
-                heightThreeStepChanged.Set();
+                if (state.Step == Step.PreVote)
+                {
+                    heightThreeStepChangedToPreVote.Set();
+                }
             };
-            // Propose -> PreVote (by consuming the message)
-            await heightThreeStepChanged.WaitAsync();
+
             // Commit ends
-            await heightTwoStepChanged.WaitAsync();
-            // GST, PreVote -> Propose (NewHeight and NewRound started)
-            await NewHeightDelayAssert(3);
+            await heightTwoStepChangedToEndCommit.WaitAsync();
             // Propose -> PreVote (message consumed)
-            await heightThreeStepChanged.WaitAsync();
+            await heightThreeStepChangedToPreVote.WaitAsync();
```

Comment:
If I understood correctly, This does not check GST time but checking change of next height step change?